src/middlewared/middlewared/api/v25_04_0/user.py

```diff
@@ -0,0 +1,107 @@
+from typing import List, Optional
+
+from annotated_types import Ge, Le
+from pydantic import EmailStr
+from typing_extensions import Annotated
+
+from middlewared.api.base import *
+
+__all__ = ["UserEntry", "UserCreateArgs", "UserCreateResult", "UserUpdateArgs", "UserUpdateResult",
+           "UserRenew2faSecretArgs", "UserRenew2faSecretResult"]
+
+
+DEFAULT_HOME_PATH = "/var/empty"
+TRUENAS_IDMAP_DEFAULT_LOW = 90000001
+
+
+class UserEntry(BaseModel):
+    id: int
+    uid: int
+    username: LocalUsername
+    unixhash: Private[str]
+    smbhash: Private[str]
+    home: NonEmptyString = DEFAULT_HOME_PATH
+    shell: NonEmptyString = "/usr/bin/zsh"
+    "Available choices can be retrieved with `user.shell_choices`."
+    full_name: NonEmptyString
+    builtin: bool
+    smb: bool = True
+    group: dict
+    groups: List[int] = []
+    """Specifies whether the user should be allowed access to SMB shares. User will also automatically be added to
+    the `builtin_users` group."""
+    password_disabled: bool = False
+    ssh_password_enabled: bool = False
+    "Required if `password_disabled` is false."
+    sshpubkey: Optional[LongString] = None
+    locked: bool = False
+    sudo_commands: List[NonEmptyString] = []
+    sudo_commands_nopasswd: List[NonEmptyString] = []
+    email: Optional[EmailStr] = None
+    id_type_both: bool
+    local: bool
+    immutable: bool
+    twofactor_auth_configured: bool
+    nt_name: Optional[str]
+    sid: Optional[str]
+    roles: List[str]
+
+
+class UserCreate(UserEntry):
+    id: excluded() = excluded_field()
```

Comment:
Flake8 complains about these types of statements:  "Call expression not allowed in type expression"